### DISCUSSION TITLE
[firebase_hosting] Wasm is not used in Flutter Web App

### DISCUSSION THREAD

**Discussion by Kuhlemann** [firebase/flutterfire#13162] [Q&A] [answered]
### Problem

Wasm is not used when deploying to Firebase Hosting and I do not understand why.

### Analysis
Deploying the same flutter code locally works with Wasm-active.
I checked all requirements which I found at: https://docs.flutter.dev/platform-integration/web/wasm

### Steps to reproduce

1. I use Flutter stable channel on version 3.24.0
2. I use the simplest possible code to reproduce (see sample below)
3. To make sure there is no problem inside the index.html I deleted the "web"-folder and recreated it with: flutter create . --platforms=web
4. flutter build web --wasm compiles successfully: "✓ Built build/web"
5. Directly after that I start a local server in my "build/web"-folder with dhttp for testing purpose: dhttpd '--headers=Cross-Origin-Embedder-Policy=credentialless;Cross-Origin-Opener-Policy=same-origin'
6. I use Chrome version 126 to open localhost:8080 and kIsWasm is **true**.
7. After that I deploy with `firebase deploy` and kIsWasm is now  **false** in the online version of my app.

What am I missing (wrong CORS configuration for example)?

```
void main() {
  runApp(
    MaterialApp(
      home: Scaffold(
        appBar: AppBar(
          title: Text('WebAssembly-Status: $kIsWasm!'),
        ),
        body: Center(
          child: Text('Welcome to Flutter WebAssembly!'),
        ),
      ),
    ),
  );
}
```

My firebase.json in my project folder looks like this:
```
{
  "hosting": {
    "source": ".",
    "ignore": [
      "firebase.json",
      "**/.*",
      "**/node_modules/**"
    ],
    "frameworksBackend": {
      "region": "europe-west1"
    },
    "rewrites": [ {
      "source": "**",
      "destination": "/index.html"
    } ],
    "headers": [
        {
           "source": "**/*",
            "headers": [
                {
                    "key": "Cross-Origin-Embedder-Policy",
                    "value": "require-corp"
                },
                {
                    "key": "Cross-Origin-Opener-Policy",
                    "value": "same-origin"
                }
            ]
        }
    ]    
  }
}
```

**Comment by shrawank039** [comment]
When you run the firebase deploy cmd in flutter. It runs flutter build web --release automatically before deployment. So as a result it rebuilds the project in release mode without the wasm support.

**Reply by shrawank039** [reply, marked as answer]
to avoid the auto build while `firebase deploy` you should set the public path in your firebase.json file.

```
{
  "hosting": {
    "public": "build/web",
    "ignore": [
      "firebase.json",
      "**/.*",
      "**/node_modules/**"
    ],
  }
}
```

changing `"source": "."` to `"public": "build/web"` the automatic build will stop on `firebase deploy`
